site/content/extend/plugins/ha.md

```diff
@@ -0,0 +1,61 @@
+---
+title: High Availabilty
+date: 2018-07-10T00:00:00-05:00
+subsection: Plugins (Beta)
+weight: 60
+---
+
+# High Availability
+
+Mattermost servers have the ability to run in High Availability (HA) mode, meaning a cluster of Mattermost app servers running together as a single Mattermost deployment. This document covers best practices for writing plugins so that they work in HA and how to deploy plugins to HA setups.
+
+## Writing Plugins
+
+It is imporant that all plugins consider HA environments when being built.
+
+Plugins are started as subprocesses of the main Mattermost process on each app server. This means a Mattermost deployment that has three app servers will have three separate copies of the same plugin running. Each running copy of the plugin will be isolated from one another on different servers. Therefore, to run properly in HA the plugin's server-side code must be stateless.
+
+To be stateless, the plugin's server-side must not retain any data in memory that might be used by the plugin in the future. Any data that needs to be retained should be stored in a place that all running copies of the plugin have access to. For example, the [key-value store](/extend/plugins/server/reference/#API.KVSet) the plugin API provides.
```

Comment:
Although your subsequent illustration helps, I think this statement is a tad too strong: it's fine to store data in memory that can be deterministically computed by each instance of the plugin (e.g. given a "channel name" configuration, cache the channel id in memory).

I'd be inclined to massage some language from the [Hypertext Transfer Protocol](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol) article on Wikipedia here:
> HTTP is a stateless protocol. A stateless protocol does not require the HTTP server to retain information or status about each user for the duration of multiple requests.